src/chatClient.cpp

```diff
@@ -1942,6 +1975,9 @@ GroupChatRoom::GroupChatRoom(ChatRoomList& parent, const mega::MegaTextChat& aCh
   (chatd::Priv)aChat.getOwnPrivilege(), aChat.getCreationTime(), aChat.isArchived()),
   mRoomGui(nullptr)
 {
+    // Get url from cache
+    mUrl = parent.mKarereClient.mChatdClient->getUrlByShard(mShardNo);
```

Comment:
Better as part of the initialization list :)
Also, pass the `url` to the `ChatRoom` ctor, since it's its member.